### ISSUE TITLE
Can the supported formats of a surface be VK_FORMAT_UNDEFINED?

### ISSUE THREAD

**Issue by tomaka** [KhronosGroup/Vulkan-Docs#207]
Two of the swapchain examples contain this code:

```
VkFormat swapchainFormat;
// If the format list includes just one entry of VK_FORMAT_UNDEFINED,
// the surface has no preferred format.  Otherwise, at least one
// supported format will be returned (assuming that the
// vkGetPhysicalDeviceSurfaceSupportKHR function, in the
// VK_KHR_surface extension returned support for the surface).
if ((formatCount == 1) && (pSurfFormats[0].format == VK_FORMAT_UNDEFINED))
    swapchainFormat = VK_FORMAT_R8G8B8_UNORM;
else
{
    assert(formatCount >= 1);
    swapchainFormat = pSurfFormats[0].format;
}
```

However the documentation of `vkGetPhysicalDeviceSurfaceFormatsKHR` doesn't mention anything about this situation where `VK_FORMAT_UNDEFINED` is returned. I don't think it's mentioned anywhere else in the specs either.

**Comment by krOoze** [comment]
Well, there is even no such function as `vkGetSurfaceFormatsKHR()`, which is referenced in the text a few times (E.2. VK_KHR_surface and E.3. VK_KHR_swapchain chapters) and used to get that UNDEFINED value, so I would assume the text and examples were just not sufficiently wholisticaly updated.

But there doesn't seem to be any explicitly stated uses for `VK_FORMAT_UNDEFINED` so it's maybe valid thing to use and receive (unless stated otherwise. It is bit funny too the "UNDEFINED", is specified to mean "unspecified".

**Comment by oddhack** [comment]
The 1.0.14 spec update has added a Note to the description of VkSurfaceFormatKHR addressing this case. There's a bit more to do that will follow in the future.

**Comment by timofonic** [comment]
@tomaka @krOoze @oddhack Any news?

**Comment by oddhack** [comment]
This should be fixed in the 1.1.111 spec update.
